tianshou/policy/modelfree/pg.py

```diff
@@ -39,30 +55,32 @@ def __init__(
         self,
         model: torch.nn.Module,
         optim: torch.optim.Optimizer,
-        dist_fn: Type[torch.distributions.Distribution],
+        dist_fn: Callable[[TDistParams], torch.distributions.Distribution],
         discount_factor: float = 0.99,
         reward_normalization: bool = False,
         action_scaling: bool = True,
-        action_bound_method: str = "clip",
+        action_bound_method: Optional[Literal["clip", "tanh"]] = "clip",
         deterministic_eval: bool = False,
         **kwargs: Any,
     ) -> None:
         super().__init__(
             action_scaling=action_scaling,
             action_bound_method=action_bound_method,
-            **kwargs
+            **kwargs,
         )
         self.actor = model
         try:
-            if action_scaling and not np.isclose(model.max_action, 1.):  # type: ignore
+            if action_scaling and not np.isclose(model.max_action, 1.0):  # type: ignore
                 import warnings
+
                 warnings.warn(
                     "action_scaling and action_bound_method are only intended"
                     "to deal with unbounded model action space, but find actor model"
                     f"bound action space with max_action={model.max_action}."
                     "Consider using unbounded=True option of the actor model,"
-                    "or set action_scaling to False and action_bound_method to \"\"."
+                    'or set action_scaling to False and action_bound_method to "".'
                 )
+        # TODO: why this try/except? warnings is a standard library module
```

Comment:
yeah looks pretty strange to me as well, not sure why did it get into the codebase... sorry!

Reply:
Np, I was just a bit afraid of touching it. I would go over the TODOs again once the build has been modernized